packages/ember-application/lib/system/application.js

```diff
@@ -456,50 +577,74 @@ var Application = Namespace.extend(RegistryProxy, {
   },
 
   /**
-    Calling initialize manually is not supported.
-
-    Please see Ember.Application#advanceReadiness and
-    Ember.Application#deferReadiness.
+    Initialize the application and return a promise that resolves with the `Ember.Application`
+    object when the boot process is complete.
 
-    @private
-    @deprecated
-    @method initialize
-   **/
-  initialize() {
-    deprecate('Calling initialize manually is not supported. Please see Ember.Application#advanceReadiness and Ember.Application#deferReadiness');
-  },
+    Run any application initializers and run the application load hook. These hooks may
+    choose to defer readiness. For example, an authentication hook might want to defer
+    readiness until the auth token has been retrieved.
 
-  /**
-    Initialize the application. This happens automatically.
+    By default, this method is called automatically on "DOM ready"; however, if autoboot
+    is disabled, you would need to call this method yourself. Calling this method before
+    "DOM ready" is not recommended.
```

Comment:
I wouldn't say it's not recommended but rather there are caveats. The caveats being that your application cannot rely on DOM readiness.